src/Illuminate/Console/Scheduling/ScheduleBackgroundCommand.php

```diff
@@ -54,8 +54,24 @@ public function __construct(Schedule $schedule)
      */
     public function handle()
     {
-        collect($this->schedule->events())->filter(function ($value) {
-            return $value->mutexName() == $this->argument('id');
-        })->each->callAfterCallbacks($this->laravel);
+        if (! $event = $this->findEventMutex()) {
+            $this->error('No scheduled event could be found that matches the given id.');
+
+            return;
+        }
+
+        $event->runCommandInForeground($this->laravel);
+    }
+
+    /**
+     * Find the event that matches the id.
+     *
+     * @return mixd
+     */
+    protected function findEventMutex()
+    {
+        return collect($this->schedule->events())->filter(function ($value) {
```

Comment:
mmmm... I dont think so? You need to filter all events against the mutex, which is a command to check.

This is similar to the command Taylor already had for `schedule:background` in 5.4 - so I was trying to keep the logic similar where possible to minimize changes.